controllers/awsmachinepool_controller.go

```diff
@@ -126,7 +134,7 @@ func (r *AWSMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 		// remove finalizer from AWSCluster
 		{
-			awsCluster, err := key.GetAWSClusterByName(ctx, r.Client, clusterName)
+			awsCluster, err := key.GetAWSClusterByName(ctx, r.Client, clusterName, awsMachinePool.GetNamespace())
```

Comment:
Isn't this controller reconciling `awsmachinepools`? How come we are adding and removing finalizers to / from `AWSClusters`?